### DISCUSSION TITLE
Questions about HB start theme

### DISCUSSION THREAD

**Discussion by bajudinasmalam** [hbstack/.github#43] [General]
i follow all the docs , but still get page not found
can you help me 
thanks

**Comment by razonyang** [comment]
Hi, would you mind explaining the error in detail? Is the error occurred in console? If so, please provide the full log.

You can also provide you site to reproduce this issue.

**Comment by bajudinasmalam** [comment]
i try start with powershell , follow tutorial on docs, but all failed . i think i must be do mistake but dont know
![Screenshot_63](https://github.com/hbstack/.github/assets/147392289/ba0449d1-1c66-44ab-9d5f-03ca20bf914b)
![Screenshot_64](https://github.com/hbstack/.github/assets/147392289/25e7c4cf-a788-420d-90f0-481e4062a047)

**Reply by razonyang** [reply]
According to the logs, there isn't `go.mod` in the site folder, as your screenshot shown, `hugo mod init` failed.

To fix it, execute `hugo mod init github.com/bajudinasmalam/start-page` again, the module path should not contain scheme.

**Reply by bajudinasmalam** [reply]
its posibble to put fixed text on search box, and how to do it :) ? thanks
![Screenshot_67](https://github.com/hbstack/.github/assets/147392289/94f32446-6987-4637-bb29-a5adc4f827a3)

**Reply by razonyang** [reply]
Sure, will rise a PR later

**Reply by razonyang** [reply]
Available since v0.4.0.

```
// params.yaml
hb:
  theme_start:
    search_input_text: "HB Framework"
```

**Reply by bajudinasmalam** [reply]
i add but not work, is there anything i need add again ?
i try with start-page : , try wit theme_start :. all not work :)

**Reply by razonyang** [reply]
Do you have [upgraded](https://hbstack.dev/modules/overview/#upgrade-modules) the theme to v0.4.0?  You can simply upgrade it by `hugo mod get -u`.

**Reply by bajudinasmalam** [reply]
i try this but got error 
![Screenshot_467](https://github.com/hbstack/.github/assets/147392289/ad79ea93-51f5-4f86-af95-683c486f020e)

**Reply by razonyang** [reply]
It should work after removing the trailing dot `.`.

```sh
hugo mod get -u github.com/hbstack/theme-start@v0.4.0
```

The following command without version also work, it will pull the latest version.

```sh
hugo mod get -u github.com/hbstack/theme-start
```

For update all modules, just use

```sh
hugo mod get -u ./...
```

**Reply by bajudinasmalam** [reply]
i got error same
Error: failed to load modules: failed to download modules: binary with name "go" not found

**Reply by razonyang** [reply]
Oh, my mistake, you have to install the [Go](https://go.dev/dl/) first, you'll need to restart the PowerShell after installation.

**Reply by bajudinasmalam** [reply]
thank you , it work perfectly, 
now i try change the icon and background
is there guide for that ?

**Reply by razonyang** [reply]
Here it is, https://hbstack.dev/themes/start/look-and-feel/

**Comment by bajudinasmalam** [comment]
how to start with light theme first ?
i try change this in css but not work [data-bs-theme="light"]

thanks

**Reply by razonyang** [reply]
> how to start with light theme first ?

No, there is no such option so far, it always to be set as `auto` to match end users system side prefered theme (dark and light).
We can treat it as a [feature request](https://github.com/hbstack/header/issues/443), but I doubt if it is really useful for users.

**Comment by bajudinasmalam** [comment]
where to put seo meta , ini param.yaml ?
thanks

**Reply by razonyang** [reply]
You can either imports the [SEO modules](https://hugomods.com/docs/seo) or create a hook to include any meta you want.

1. Put the following configuration in params.yaml

```
hugopress:
  modules:
    custom:
      hooks:
        head-end:
          disable: false
```

2. And then create `layouts/partials/hugopress/modules/custom/hooks/head-end.html` in your site.

```html
<meta name="foo" content="bar">
```

3. Restart server or rebuil site, you will see the foobar meta.

**Reply by bajudinasmalam** [reply]
when i view-page-source i not found the foo meta, maybe i do it wrong
![Screenshot_470](https://github.com/hbstack/.github/assets/147392289/f868eb3e-d4cc-4810-ad38-375b5735237b)

**Reply by razonyang** [reply]
Hi, you params configuration is incorrect, the `hugopress` is same level as `hb`.

```
hb:
  ...

hugopress:
  ...
```

**Reply by bajudinasmalam** [reply]
i do it same, but why still cant, can you help me 
thanks

**Reply by razonyang** [reply]
The filename is also incorrect, it should be layouts/partials/hugopress/modules/custom/hooks/head-end.html.
It will save your time if you read the comments again and check yours.

**Reply by bajudinasmalam** [reply]
im so sorry, 
thank you, it works
i can add any meta at head-end.html right ?

**Reply by razonyang** [reply]
No problem.

Sure you can add anything. For complicated usage, please read Hugo template syntax, you can access the page variable via `.Page` in that template.

**Comment by bajudinasmalam** [comment]
how to add schema ?
thanks

**Reply by razonyang** [reply]
https://hugomods.com/docs/seo/

**Reply by bajudinasmalam** [reply]
have add hugo mod get https://hugomods.com/docs/seo/
but where and how put the schema code ?
thanks

**Reply by razonyang** [reply]
You should import the specified module on that page, in this case, it's `github.com/hugomods/seo/modules/schema`.

To import a module, you'll need to modify the module.yaml (default to `config/_default/module.yaml` in the exampleSite)

```
imports:
# ... other modules
- path: github.com/hugomods/seo/modules/schema
```

The theme support HugoPress, that's mean, you only need import those SEO modules, then the module will work.
